### PR TITLE
Part Design: Improved chamfer behaviour when faces are selected and non equal chamfer types are used

### PR DESCRIPTION
**Pull request by troyp76**
Improved Part Design chamfer behaviour when faces are selected and non equal chamfer types are used. 
(Distance Distance or Distance Angle)

See  [Forum discussion](https://forum.freecadweb.org/viewtopic.php?f=19&t=62084) for background.

_This is resubmission of #5029 with clean commit history and review feedback addressed._

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [X]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [X]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [X] Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [X]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [X]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [X]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [X]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/PartDesign/App/FeatureChamfer.cpp
Comment (chennes) [inline src/Mod/PartDesign/App/FeatureChamfer.cpp:146]:
```suggestion
        for (const auto &itSN : SubNames) {
```

File: src/Mod/PartDesign/App/FeatureChamfer.cpp
Comment (chennes) [inline src/Mod/PartDesign/App/FeatureChamfer.cpp:160]:
```suggestion
                for (const auto &itFN : FaceNames) {
```

File: src/Mod/PartDesign/App/FeatureChamfer.cpp
Comment (chennes) [inline src/Mod/PartDesign/App/FeatureChamfer.cpp:168]: Totally inconsequential, but we typically don't use the curly braces on single-line conditionals (and that appears to be the usage in the rest of this file). So perhaps:
```suggestion
                    if ( faceLast.IsEqual(selFace) )
                        face = TopoDS::Face( flipDirection ? faceFirst : faceLast );
                    else if ( faceFirst.IsEqual(selFace) )
                        face = TopoDS::Face( flipDirection ? faceLast : faceFirst );
```

**Review by chennes — APPROVED** [review]
Nice PR, things are looking good here. It compiles and runs, and my testing didn't reveal any problems, it behaved exactly as expected (unlike the existing code!). I've added a few very minor comments, but none are merge-blockers. I'll wait for some others to test before merging it, though.

File: src/Mod/PartDesign/App/FeatureChamfer.cpp
Comment (troyp76) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp:168]: @chennes 
Thanks for the feedback. I might be showing my age by saying this but the last time I did any serious C++ other than maintenance was back in the late 90's so C++98 level. Looks like need to do a refresher!

Happy to make the changes that you suggested, but given my rather embarrassing experience with the last PR I wanted to ask if it's best to accept the changes directly here in GH or change it locally and push a new commit?

Comment (chennes) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp:168]: I can only tell you my personal preference when coding, which is to make the changes by hand on my local branch, commit locally, then squash the new commit and force-push the branch. That way I keep the PR to a single commit, and can test the patches before committing them (reviewers make mistakes!). However, you are welcome to accept the suggestions via GitHub if you want to try that workflow instead.

Comment (troyp76) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp:168]: The local commit / squash / force push approach makes sense.
Will wait for further feedback and testing.

Thanks for taking the time to explain. Cheers.

Comment (chennes) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp:168]: I'd say that GeneFC's testing was pretty thorough, I'd say go ahead and do a final cleanup path and let's merge it.

File: src/Mod/PartDesign/App/FeatureChamfer.cpp
Comment (troyp76) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp]: Changed made as suggested.

File: src/Mod/PartDesign/App/FeatureChamfer.cpp
Comment (troyp76) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp]: Changed made as suggested.

**Comment by troyp76** [comment]
@chennes 
Suggested changes have been made and committed to the branch.
Ready to merge.

**Comment by chennes** [comment]
Excellent, thanks for the PR and quick revisions.

**Comment by luzpaz** [comment]
@troyp76 thanks for the PR. Some late feedback: please remember next time to prepend the commit summary with the name of the workbench you're submitting code to. In this case it was `PartDesign:` or `PD:` Thanks in advance!

**Comment by troyp76** [comment]
@luzpaz 
Thanks for the heads up, I thought it was only meant to be the PR itself.

Is it possible to change it now or is it too late?

**Comment by luzpaz** [comment]
@troyp76 I think there is but i've never seen a dev with commit rights do it in this project. It's all good though :wink:

**Pull request merged**
